datumaro/plugins/celeba_format.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (C) 2021 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import os.path as osp
+
+from datumaro.components.annotation import (
+    AnnotationType, Bbox, Label, LabelCategories, Points,
+)
+from datumaro.components.extractor import DatasetItem, Importer, SourceExtractor
+from datumaro.util.image import find_images
+
+
+class CelebaPath:
+    IMAGES_DIR = 'Img/img_celeba'
+    IMAGES_ALIGN_DIR = 'Img/img_align_celeba'
+    LABELS_FILE = 'identity_CelebA'
+    BBOXES_FILE = 'list_bbox_celeba.txt'
+    ATTRS_FILE = 'list_attr_celeba.txt'
+    LANDMARKS_FILE = 'list_landmarks_celeba.txt'
+    LANDMARKS_ALIGN_FILE = 'list_landmarks_align_celeba.txt'
+    SUBSETS_FILE = 'Eval/list_eval_partition.txt'
+    SUBSETS = {'0': 'train', '1': 'val', '2': 'test'}
+
+class CelebaExtractor(SourceExtractor):
+    def __init__(self, path, subset=None):
+        if not osp.isfile(path):
+            raise FileNotFoundError("Can't read annotation file '%s'" % path)
+
+        super().__init__()
+        self._anno_dir = osp.dirname(path)
+
+        self._categories = { AnnotationType.label: LabelCategories() }
+        self._items = list(self._load_items(path).values())
+
+    def _load_items(self, path):
+        items = {}
+
+        image_dir = osp.join(osp.dirname(self._anno_dir), CelebaPath.IMAGES_DIR)
+        if not osp.isdir(image_dir):
+            image_dir = osp.join(osp.dirname(self._anno_dir), CelebaPath.IMAGES_ALIGN_DIR)
+        if osp.isdir(image_dir):
+            images = {
+                osp.splitext(osp.relpath(p, image_dir))[0].replace('\\', '/'): p
+                for p in find_images(image_dir, recursive=True)
+            }
+        elif osp.isdir(osp.join(osp.dirname(self._anno_dir), CelebaPath.IMAGES_DIR)):
+            images = {}
+
+        label_categories = self._categories[AnnotationType.label]
+        with open(path, encoding='utf-8') as f:
+            for line in f:
+                item_id, item_ann = self.split_annotation(line)
+                label_ids = [int(id) for id in item_ann]
+                anno = []
+                for label in label_ids:
+                    if label > len(label_categories):
+                        for i in range(len(label_categories), label + 1):
+                            label_categories.add('label_%d' % i)
+                    anno.append(Label(label))
+
+                items[item_id] = DatasetItem(id=item_id, image=images.get(item_id),
+                    annotations=anno)
+
+        bbox_path = osp.join(self._anno_dir, CelebaPath.BBOXES_FILE)
+        if osp.isfile(bbox_path):
+            with open(bbox_path, encoding='utf-8') as f:
+                for line in f:
+                    if len(line.split()) < 5 or 'image_id' == line[0:8]:
+                        continue
```

Comment:
I presume this is meant to skip the two initial lines, but you shouldn't skip them. The first line contains the number of entries, which you should use to verify the integrity of the file at the end; and the second line contains the table header which you should compare against the expected one to make sure that the file has the right format.